app.py

```diff
@@ -0,0 +1,26 @@
+from graph import *
+import random
+from prim import *
+
+
+def generate_weights(edges):
+    weights = [[float('inf') for i in range(N)] for j in range(N)]
```

Comment:
why implmant graph as a 'Adjacency List' if  you create  weights  in 'Adjacency Matrix' ? 

i think that the bast way is to add to en adge proprty 'weight'

Reply:
לפי הדרישה האלגוריתם מקבל גרף ופונקציית משקל (בנפרד)
זה עוזר אם רוצים להתאים סט משקלים שונה או להשתמש בגרף לא ממושקל, אבל זה לא באמת משנה

Reply:
אין בעיה, אבל התעלמת מהחלק הראשון (והחשוב) של התגובה, למה את יוצרת פה מטריצה ? לא עדיף לייצג ברשימת קשירות?

Reply:
לדעתי לא
רשימת קשירויות נוחה השרותים לעבור ברצף על כל המשחקים של צומת מסוים ואין לנו צורך כזה
נעדיף גישה מיידית O(1) למשקל בתא מסוים במטריצה

Reply:
קיבלתי